docs/recipes/LocalFirstDataWithPowerSync.md

```diff
@@ -0,0 +1,2037 @@
+:::
+
+### Authenticating with Supabase
+
+PowerSync needs a valid session token to connect to the Supabase, so we'll need a hook and context to manage our authentication state.
+
+This is a basic example that uses a context to manage the authentication state and a hook to access that state in your components.
+
+Add the following code to `app/services/database/use-auth.tsx`:
+
+```tsx
+// app/services/database/use-auth.tsx
+import {User} from "@supabase/supabase-js"
+import {supabase} from "app/services/database/supabase"
+import React, {createContext, PropsWithChildren, useCallback, useContext, useMemo, useState} from "react"
+
+type AuthContextType = {
+  signIn: (email: string, password: string) => void
+  signUp: (email: string, password: string) => void
+  signOut: () => void
+  signedIn: boolean
+  loading: boolean
+  error: string
+  user: User | null
+}
+
+// We initialize the context with null to ensure that it is not used outside of the provider
+const AuthContext = createContext<AuthContextType | null>(null)
+
+/**
+ * AuthProvider manages the authentication state and provides the necessary methods to sign in, sign up and sign out.
+ */
+export const AuthProvider = ({children}: PropsWithChildren<any>) => {
+  const [signedIn, setSignedIn] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState("")
+  const [user, setUser] = useState<User | null>(null)
+
+
+  // Sign in with provided email and password
+  const signIn = useCallback(async (email: string, password: string) => {
+    setLoading(true)
+    setError("")
+    setUser(null)
+    try {
+      const {data: {session, user}, error} = await supabase.auth.signInWithPassword({email, password})
+      if (error) {
+        setSignedIn(false)
+        setError(error.message)
+      } else if (session && user) {
+        setSignedIn(true)
+        setUser(user)
+      }
+    } catch (error: any) {
+      setError(error?.message ?? "Unknown error")
+    } finally {
+      setLoading(false)
+    }
+  }, [
+    setSignedIn, setLoading, setError, setUser, supabase
+  ])
+
+  // Create a new account with provided email and password
+  const signUp = useCallback(async (email: string, password: string) => {
+    setLoading(true)
+    setError("")
+    setUser(null)
+    try {
+      const {data, error} = await supabase.auth.signUp({email, password})
+      if (error) {
+        setSignedIn(false)
+        setError(error.message)
+      } else if (data.session) {
+        await supabase.auth.setSession(data.session)
+        setSignedIn(true)
+        setUser(data.user)
+      }
+    } catch (error: any) {
+      setUser(null)
+      setSignedIn(false)
+      setError(error?.message ?? "Unknown error")
+    } finally {
+      setLoading(false)
+    }
+  }, [
+    setSignedIn, setLoading, setError, setUser, supabase
+  ])
+
+  // Sign out the current user
+  const signOut = useCallback(async () => {
+    setLoading(true)
```

Comment:
We need to disconnect from syncing here. Ref: https://github.com/powersync-ja/powersync-js/blob/cef404919d7c1f700cb779ef430f8d4845221bd4/demos/react-native-supabase-todolist/app/views/signout.tsx#L12